python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -0,0 +1,225 @@
+
+  private static final String IP_PROTOCOL = "ip_protocol";
+  private static final String CIDR_IP = "cidr_ip";
+  private static final String CIDR_IPV6 = "cidr_ipv6";
+  private static final String IPPROTOCOL = "ipProtocol";
+  private static final String CIDRIP = "cidrIp";
+  private static final String CIDRIPV6 = "cidrIpv6";
+
+
+  private static final Set<String> SENSITIVE_PROTOCOL = Set.of("tcp", "6");
+  private static final String ANY_PROTOCOL = "-1";
+  private static final String EMPTY_IPV4 = "0.0.0.0/0";
+  private static final String EMPTY_IPV6 = "::/0";
+  private static final long[] ADMIN_PORTS = new long[]{22, 3389};
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_ec2.CfnSecurityGroup", UnrestrictedAdministrationCheck::checkCfnSecurityGroup);
+    checkFqn("aws_cdk.aws_ec2.CfnSecurityGroupIngress", UnrestrictedAdministrationCheck::checkCfnSecurityGroupIngress);
+  }
+
+  // Checks methods
+  private static void checkCfnSecurityGroup(SubscriptionContext ctx, CallExpression callExpression) {
+    getArgument(ctx, callExpression, "security_group_ingress")
+      .flatMap(flow -> flow.getExpression(isListLiteral()).map(ListLiteral.class::cast))
+      .ifPresent(list -> list.elements().expressions()
+        // Process each expression in the Python list
+        .forEach(expression -> {
+          CdkUtils.ExpressionFlow flow = CdkUtils.ExpressionFlow.build(ctx, expression);
+          raiseIssueIfIngressPropertyCallWithSensitiveArgument(ctx, flow.getLast());
+          raiseIssueIfDictionaryWithSensitiveArgument(ctx, flow.getLast());
+        })
+      );
+  }
+
+  private static void checkCfnSecurityGroupIngress(SubscriptionContext ctx, CallExpression callExpression) {
+    checkCallCfnSecuritySensitive(new Call(ctx, callExpression));
+  }
+
+  // Methods to handle call expressions
+  private static void raiseIssueIfIngressPropertyCallWithSensitiveArgument(SubscriptionContext ctx, Expression expression) {
+    getCall(expression, "aws_cdk.aws_ec2.CfnSecurityGroup.IngressProperty")
+      .map(callExpression -> new Call(ctx, callExpression))
+      .ifPresent(UnrestrictedAdministrationCheck::checkCallCfnSecuritySensitive);
+  }
+
+  private static void checkCallCfnSecuritySensitive(Call call) {
+    if (isCallWithArgumentBadProtocolEmptyIpAddressAdminPort(call) || isCallWithArgumentInvalidProtocolEmptyIpAddress(call)) {
+      getArgument(call.ctx, call.callExpression, CIDR_IP).ifPresent(flow -> flow.addIssue(MESSAGE));
+      getArgument(call.ctx, call.callExpression, CIDR_IPV6).ifPresent(flow -> flow.addIssue(MESSAGE));
+    }
+  }
+
+  private static boolean isCallWithArgumentBadProtocolEmptyIpAddressAdminPort(Call call) {
+    return call.hasArgument(IP_PROTOCOL, isString(SENSITIVE_PROTOCOL))
+      && (call.hasArgument(CIDR_IP, isString(EMPTY_IPV4)) || call.hasArgument(CIDR_IPV6, isString(EMPTY_IPV6)))
+      && call.inInterval("from_port", "to_port", ADMIN_PORTS);
```

Comment:
This method name does not feel intuitive. I would suggest something like `hasSensitvePortRange`